### DISCUSSION TITLE
Supported XCode and Command Line Tools Version

### DISCUSSION THREAD

**Discussion by MAUIoxo** [dotnet/maui#24301] [Q&A]
Which Versions of XCode and the Command Line Tools are currently supported by MAUI? I am asking because when using **DevExpress** in version **24.1.4** they do not support **Trimming** and to set trimming to **false**

`<PublishTrimmed>false</PublishTrimmed>` and compile for iOS, I get the error `"iOS projects must build with PublishTrimmed=true. Current value: false. Set 'MtouchLink=None' instead to disable trimming for all assemblies"`. 

When I do this and set

`<MtouchLink>None</MtouchLink>`, I get the error `"This version of Microsoft.iOS requires the iOS 17.5 SDK (shipped with Xcode 15.4). Either upgrade Xcode to get the required header files or set the managed linker behaviour to Link Framework SDKs Only in your project's iOS Build Options > Linker Behavior (to try to avoid the new APIs)."`.

So it looks like it is needed to have a higher version of Xcode installed than just XCode 15.2 with Command Line Tools 15.1, but I thought MAUI does not even fully support XCode 15.3 at the moment?

@jfversluis what is the current answer for this right now?

**Comment by jfversluis** [comment]
This should help: https://github.com/dotnet/maui/wiki/Release-Versions

**Reply by MAUIoxo** [reply]
@jfversluis - Thanks for this link, but is there information about the XCode versions? Can't see information about that via this link.
Or is it like it doesn't matter as long as I just use the mentioned MAUI Versions in my .csproj, e.g. `<MauiVersion>8.0.70</MauiVersion>` and then any higher versions just compile for that and its supported?

**Reply by jfversluis** [reply]
The tables can scroll to the right :) supported Xcode versions are there!

Xcode versions can definitely matter you probably want to hold off on the newest version until you know it works well with our stuff. The above link will keep getting updated with that information.

Right now the latest stable version, 15.4, should work fine

**Reply by MAUIoxo** [reply]
Oh my goodness - sorry for asking again, I didn't see that it is scrollable. Thanks now, I have the right information!
